rest/config.go

```diff
@@ -472,6 +472,29 @@ func (dbConfig *DbConfig) UseXattrs() bool {
 	return base.DefaultUseXattrs
 }
 
+// Create a deepcopy of this DbConfig, or panic.
+// This will only copy all of the _exported_ fields of the DbConfig.
+func (dbConfig *DbConfig) MustDeepCopy() (dbConfigCopy *DbConfig) {
+
+	dbConfigDeepCopy := &DbConfig{}
+	err := base.DeepCopy(&dbConfigDeepCopy, dbConfig)
+	if err != nil {
+		base.Panicf(base.KeyAll, "Error trying to peform deep copy of DbConfig.  Err: %v" , err)
+	}
+	return dbConfigDeepCopy
+
+}
+
+// Implementation of AuthHandler interface for ShadowConfig
```

Comment:
ShadowConfig and ChannelIndexConfig should already support GetCredentials via BucketConfig.

Reply:
Not sure how/why those made it into the PR.  Fixed